src/three/math/Ellipsoid.js

```diff
@@ -200,4 +203,43 @@ export class Ellipsoid {
 
 	}
 
+	calculateHorizonDistance( latitude, elevation ) {
+
+		// elevation relative to the surface of the ellipsoid at the specified latitude
+		const effectiveRadius = this.calculateEffectiveRadius( latitude );
+		return Math.sqrt( 2 * effectiveRadius * elevation + elevation ** 2 );
+
+	}
+
+	calculateEffectiveRadius( latitude ) {
+
+		const semiMajorAxis = this.radius.x;
+		const semiMinorAxis = this.radius.z;
+		const eSquared = 1 - ( semiMinorAxis ** 2 / semiMajorAxis ** 2 );
+		const phi = latitude * MathUtils.DEG2RAD;
+
+		const sinPhiSquared = Math.sin( phi ) ** 2;
+		const N = semiMajorAxis / Math.sqrt( 1 - eSquared * sinPhiSquared );
+		return N;
+
+	}
```

Comment:
Are these formulas from somewhere? Can we add more comments?

Reply:
I added source for the formulas